### PR TITLE
[DROOLS-7284] Bump apache-karaf from 4.3.6 to 4.3.8 in /kie-osgi/kie-…

### PR DESCRIPTION
**Pull request by tkobayas**
…karaf-itests

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7284

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
GHA Java 8:
```
2023-01-19T03:50:20.7831347Z [ERROR] Tests run: 15, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 15.726 s <<< FAILURE! - in org.kie.server.client.LoadBalancerClientTest
2023-01-19T03:50:20.7832014Z [ERROR] testDefaultLoadBalancerNoServersAvailable(org.kie.server.client.LoadBalancerClientTest)  Time elapsed: 5.181 s  <<< ERROR!
```
LoadBalancerClientTest is a flaky test but,

kie-karaf-itest also fails.
```
2023-01-19T05:45:58.8253725Z [INFO] Running org.kie.karaf.itest.blueprint.KieBlueprintDependencyKarafIntegrationTest
2023-01-19T05:48:59.9660108Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 181.128 s <<< FAILURE! - in org.kie.karaf.itest.blueprint.KieBlueprintDependencyKarafIntegrationTest
2023-01-19T05:48:59.9661097Z [ERROR] testKieBase(org.kie.karaf.itest.blueprint.KieBlueprintDependencyKarafIntegrationTest)  Time elapsed: 180.802 s  <<< ERROR!
2023-01-19T05:48:59.9661861Z java.rmi.NotBoundException: 325db55b-95e8-47fe-a36e-89e2cd8c9235
...
2023-01-19T07:04:19.4014486Z [ERROR] Tests run: 42, Failures: 0, Errors: 42, Skipped: 0
```

I'm going to investigate...

File: kie-osgi/kie-karaf-features/src/main/filtered-resources/repository/features-core.xml [hunk 1]
```diff
@@ -104,9 +104,9 @@
     <feature>camel-core</feature>
     <feature>camel-spring</feature>
     <feature>camel-cxf</feature>
-    <feature version="[3.3,3.5)">cxf-specs</feature>
-    <feature version="[3.3,3.5)">cxf-core</feature>
-    <feature version="[3.3,3.5)">cxf-jaxrs</feature>
+    <feature version="[3.3,3.6)">cxf-specs</feature>
+    <feature version="[3.3,3.6)">cxf-core</feature>
+    <feature version="[3.3,3.6)">cxf-jaxrs</feature>
```
Comment (mareknovotny) [inline kie-osgi/kie-karaf-features/src/main/filtered-resources/repository/features-core.xml:109]: so if you extend the range for cxf-* deps, shouldn't be changed also kie-parent version property at https://github.com/kiegroup/droolsjbpm-build-bootstrap/commit/bbb34b2be08cedd09f93fcca7af0b8f2110dc473#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R316

Comment (tkobayas) [reply kie-osgi/kie-karaf-features/src/main/filtered-resources/repository/features-core.xml:109]: > so if you extend the range for cxf-* deps, shouldn't be changed also kie-parent version property at [kiegroup/droolsjbpm-build-bootstrap@bbb34b2#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R316](https://github.com/kiegroup/droolsjbpm-build-bootstrap/commit/bbb34b2be08cedd09f93fcca7af0b8f2110dc473#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R316)

@mareknovotny I'm afraid if I misunderstand your comment. You mean I should change the `droolsjbpm-build-bootstrap/pom.xml`? No, I changed the version range in the `features-core.xml` to meet the current `droolsjbpm-build-bootstrap/pom.xml` 's `version.org.apache.cxf` value `3.5.5` in order to pass the tests.

**Comment by grgrzybek** [comment]
Roughly the change looks fine. But give me a bit of time to catch up with the changes ;)
Also - I'll check it together with https://issues.redhat.com/browse/BAPL-2369 (or rather check this PR when starting to work on BAPL-2369).

File: kie-osgi/kie-karaf-features/src/main/filtered-resources/repository/features-core.xml (hunk 1 above)
Comment (mareknovotny) [reply kie-osgi/kie-karaf-features/src/main/filtered-resources/repository/features-core.xml:109]: yes, currently kie-parent pom has 3.5.5 and i saw that 3.6, but I didn't realize it is less than 3.6.

**Comment by tkobayas** [comment]
GHA Java 8:

LoadBalancerClientTest is a flaky test.
```
2023-01-19T09:55:01.4520709Z [ERROR] Tests run: 15, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 16.573 s <<< FAILURE! - in org.kie.server.client.LoadBalancerClientTest
2023-01-19T09:55:01.4521700Z [ERROR] testDefaultLoadBalancerNoServersAvailable(org.kie.server.client.LoadBalancerClientTest)  Time elapsed: 5.192 s  <<< ERROR!
2023-01-19T09:55:01.4522626Z org.kie.server.common.rest.NoEndpointFoundException: No available endpoints found
```

kie-karaf-itests passed.
```
2023-01-19T12:01:55.1681733Z [INFO] --- maven-failsafe-plugin:2.22.1:integration-test (default) @ kie-karaf-itests ---
2023-01-19T12:01:55.1727291Z [INFO] Failsafe report directory: /home/runner/work/droolsjbpm-integration/droolsjbpm-integration/kiegroup_droolsjbpm_integration/kie-osgi/kie-karaf-itests/target/failsafe-reports
2023-01-19T12:01:55.1747891Z [INFO]
2023-01-19T12:01:55.1748539Z [INFO] -------------------------------------------------------
2023-01-19T12:01:55.1749096Z [INFO]  T E S T S
2023-01-19T12:01:55.1750070Z [INFO] -------------------------------------------------------
2023-01-19T12:01:55.6549503Z [INFO] Running org.kie.karaf.itest.blueprint.KieBlueprintDependencyKarafIntegrationTest
2023-01-19T12:02:17.7845907Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 22.126 s - in org.kie.karaf.itest.blueprint.KieBlueprintDependencyKarafIntegrationTest
...
2023-01-19T12:18:36.5534576Z [WARNING] Tests run: 62, Failures: 0, Errors: 0, Skipped: 2
```

**Comment by tkobayas** [comment]
@grgrzybek Thank you for reviewing! Btw, if you want some changes for BAPL-2369, we can do it in another PR :)

**Review by grgrzybek — APPROVED** [review]
Approving the Karaf upstream part, and I'll do actual alignment for `-Pfuse` part, where other set of features is used (for BAPL-2369), ok?

**Comment by tkobayas** [comment]
> I'll do actual alignment for `-Pfuse` part, where other set of features is used (for BAPL-2369), ok?

Yes, thanks, @grgrzybek !

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**